src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -795,14 +805,15 @@ private SqlCommand BuildGetChangesCommand(SqlConnection connection, SqlTransacti
                     c.SYS_CHANGE_OPERATION,
                     l.{LeasesTableChangeVersionColumnName},
                     l.{LeasesTableAttemptCountColumnName},
-                    l.{LeasesTableLeaseExpirationTimeColumnName}
+                    l.{LeasesTableLeaseExpirationTimeColumnName},
+                    CASE WHEN (l.{LeasesTableLeaseExpirationTimeColumnName} IS NULL OR l.{LeasesTableLeaseExpirationTimeColumnName} < SYSDATETIME()) THEN 0 else 1 END AS IsLeaseLocked
                 FROM CHANGETABLE(CHANGES {this._userTable.BracketQuotedFullName}, @last_sync_version) AS c
                 LEFT OUTER JOIN {this._leasesTableName} AS l ON {leasesTableJoinCondition}
                 LEFT OUTER JOIN {this._userTable.BracketQuotedFullName} AS u ON {userTableJoinCondition}
                 WHERE
                     (l.{LeasesTableLeaseExpirationTimeColumnName} IS NULL AND
                        (l.{LeasesTableChangeVersionColumnName} IS NULL OR l.{LeasesTableChangeVersionColumnName} < c.{SysChangeVersionColumnName}) OR
-                        l.{LeasesTableLeaseExpirationTimeColumnName} < SYSDATETIME()
+                        l.{LeasesTableLeaseExpirationTimeColumnName} IS NOT NULL
```

Comment:
We only select the max batch size in this query, so you won't be getting the full number of locked rows.

We'll probably want to just do a separate query to get the locked lease count. That also means we don't have to get the row data for the locked rows (which isn't necessary and could impact performance)

Reply:
get only the lease locked rows count in a separate query wouldn't impact the performance, I'll go ahead and update to that.